sympy/functions/elementary/hyperbolic.py

```diff
@@ -229,6 +229,19 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_is_real(self):
+        arg = self.args[0]
+        if arg.is_real:
+            return True
+
+        # if `im` is of the form n*pi
+        # else, check if it is a number
+        re, im = arg.as_real_imag()
+        if (im % pi) == 0:
```

Comment:
What if the condition is undecided?

Reply:
I think maybe `return (im % pi).is_zero` should work for all cases.

There are many of cases like this in the assumptions code which makes me wonder whether we need a specific modulo assumption type method/function (`is_divisible_my` or something) rather than evaluating new expressions in the assumptions code.